### PR TITLE
Tag interfering with Xpath fix

### PR DESCRIPTION
**Pull request by KhoomeiK**
Inserting the tag while calculating the xpaths for elements was resulting in the tarsier tags interfering with xpath indexing for spans. I separated tag insertion out to a separate loop that runs after xpaths are calculated.

### CODE REVIEW

File: tarsier/tag_utils.ts [hunk 1]
```diff
@@ -162,36 +162,69 @@ window.tagifyWebpage = (tagLeafTexts = false) => {
       continue;
     }
 
-    const intractable = isInteractable(el);
+    const interactable = isInteractable(el);
     const elTagName = el.tagName.toLowerCase();
-    const idStr = inputTags.includes(elTagName) ? `{${idNum}}` : `[${idNum}]`;
+    // const idStr = inputTags.includes(elTagName) ? `{${idNum}}` : `[${idNum}]`;
```
Comment (asim-shrestha) [inline tarsier/tag_utils.ts:167]: Can we jus delete the commented code?

File: tarsier/tag_utils.ts [hunk 2]
```diff
@@ -162,36 +162,69 @@ window.tagifyWebpage = (tagLeafTexts = false) => {
...
     idToXpath[idNum] = getElementXPath(el);
 
+    // create the span for the id tag
+    // let idSpan = create_tagged_span(idStr);
```
Comment (awtkns) [inline tarsier/tag_utils.ts]: We should remove all commented out code

File: tarsier/tag_utils.ts [hunk 3]
```diff
@@ -162,36 +162,69 @@ window.tagifyWebpage = (tagLeafTexts = false) => {
...
+    if (isEmpty(el) || !elIsClean(el)) {
+      continue;
+    }
+
+    const interactable = isInteractable(el);
+    const elTagName = el.tagName.toLowerCase();
+    const idStr = inputTags.includes(elTagName) ? `{${idNum}}` : `[${idNum}]`;
+
     // create the span for the id tag
     let idSpan = create_tagged_span(idStr);
```
Comment (awtkns) [inline tarsier/tag_utils.ts:198]: this should be a const

File: tarsier/tag_utils.ts [hunk 4]
```diff
@@ -162,36 +162,61 @@ window.tagifyWebpage = (tagLeafTexts = false) => {
...
 
+    if (interactable) {
+      if (!inputTags.includes(elTagName)) {
+        idNum++;
+      } else if (elTagName === "textarea" || elTagName === "input") {
+        idNum++;
+      } else if (elTagName === "select") {
+        // leave select blank - we'll give a tag ID to the options
+      }
+    } else {
```
Comment (awtkns) [inline tarsier/tag_utils.ts:177]: can this be simplified? This can be combined with the statement above

File: tarsier/tag_utils.ts [hunk 5]
```diff
@@ -162,36 +162,69 @@ window.tagifyWebpage = (tagLeafTexts = false) => {
       continue;
     }
```
Comment (asim-shrestha) [inline tarsier/tag_utils.ts:164]: Functions a bit long and hard to read. Ideally we broke this apart but given this is a js utility file in a python project this should be fine.

**Review by awtkns — APPROVED** [review]

**Pull request merged**